modules/search/instant-search/components/search-app.jsx

```diff
@@ -43,40 +40,43 @@ class SearchApp extends Component {
 			showResults: false,
 		};
 		this.getResults = debounce( this.getResults, 200 );
-		this.prepareDomForMounting();
 	}
 
 	componentDidMount() {
 		this.getResults( { sort: this.props.initialSort } );
-
 		this.getResults.flush();
 
+		this.addEventListeners();
+
 		if ( this.hasActiveQuery() ) {
 			this.showResults();
 		}
 
 		if ( this.props.grabFocus ) {
 			this.input.current.focus();
 		}
+	}
 
+	componentWillUnmount() {
+		this.removeEventListeners();
+	}
+
+	addEventListeners() {
 		window.addEventListener( 'popstate', this.onChangeQueryString );
 		window.addEventListener( 'queryStringChange', this.onChangeQueryString );
+
+		document
+			.querySelectorAll( this.props.themeOptions.searchInputSelector )
+			.forEach( input => input.addEventListener( 'focus', this.showResults ) );
 	}
 
-	componentWillUnmount() {
+	removeEventListeners() {
 		window.removeEventListener( 'popstate', this.onChangeQueryString );
 		window.removeEventListener( 'queryStringChange', this.onChangeQueryString );
-	}
 
-	prepareDomForMounting() {
-		// Clean up the page prior to mounting component
-		hideElements( this.props.themeOptions.elementSelectors );
-		document
-			.querySelectorAll( '.jetpack-instant-search-wrapper' )
-			.forEach( widget => removeChildren( widget ) );
 		document
-			.querySelectorAll( this.props.themeOptions.searchFormSelector )
-			.forEach( searchForm => removeChildren( searchForm ) );
+			.querySelectorAll( '.jetpack-instant-search-wrapper input.search-field' )
```

Comment:
```suggestion
			.querySelectorAll( this.props.themeOptions.searchInputSelector )
```

I forgot to update this line when adding the searchInputSelector.